animata/button/hero-button.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useCallback, useState } from "react";
+
+export default function DemoButton() {
+  const [isHovered, setIsHovered] = useState(false);
+
+  const handleMouseEnter = useCallback(() => {
+    setIsHovered(true);
+  }, []);
+
+  const handleMouseLeave = useCallback(() => {
+    setIsHovered(false);
+  }, []);
+
```

Comment:
Instead of tracking hover like this, you can use `group` hover from tailwindcss. https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state

Reply:
okay sir will do the changes by evening